dvc/command/check_ignore.py

```diff
@@ -14,27 +15,71 @@ def __init__(self, args):
         self.ignore_filter = self.repo.tree.dvcignore
 
     def _show_results(self, result):
-        if result.match or self.args.non_matching:
-            if self.args.details:
-                logger.info("{}\t{}".format(result.patterns[-1], result.file))
-            else:
-                logger.info(result.file)
+        if not result.match and not self.args.non_matching:
+            return
 
-    def run(self):
-        if self.args.non_matching and not self.args.details:
-            raise DvcException("--non-matching is only valid with --details")
+        if self.args.details:
+            patterns = result.patterns
+            if not self.args.all:
+                patterns = patterns[-1:]
 
-        if self.args.quiet and self.args.details:
-            raise DvcException("cannot both --details and --quiet")
+            for pattern in patterns:
+                logger.info("{}\t{}".format(pattern, result.file))
+        else:
+            logger.info(result.file)
+
+    def _check_one_file(self, target):
+        result = self.ignore_filter.check_ignore(target)
+        self._show_results(result)
+        if result.match:
+            return 0
+        return 1
 
+    def _interactive_mode(self):
+        ret = 1
+        while True:
+            target = ask("")
+            if target == "":
+                logger.info(
+                    "Empty string is not a valid pathspec. Please use . "
+                    "instead if you meant to match all paths."
+                )
+                break
+            if not self._check_one_file(target):
+                ret = 0
+        return ret
+
+    def _normal_mode(self):
         ret = 1
         for target in self.args.targets:
-            result = self.ignore_filter.check_ignore(target)
-            self._show_results(result)
-            if result.match:
+            if not self._check_one_file(target):
                 ret = 0
         return ret
 
+    def _check_args(self):
+        if not self.args.stdin and not self.args.targets:
+            raise DvcException("`targets` or `--stdin` needed")
+
+        if self.args.stdin and self.args.targets:
+            raise DvcException("cannot have both `targets` and `--stdin`")
+
+        if self.args.non_matching and not self.args.details:
+            raise DvcException(
+                "`--non-matching` is only valid with `--details`"
+            )
+
+        if self.args.all and not self.args.details:
+            raise DvcException("`--all` is only valid with `--details`")
+
+        if self.args.quiet and self.args.details:
+            raise DvcException("cannot use both `--details` and `--quiet`")
```

Comment:
Maybe --quiet shouold just win here?

Reply:
> Maybe --quiet should just win here?

Yes, it used to be `--verbose` and `--quiet`. They might cause some problems, and had been fixed in #4304.
And for `--details` and `--quiet` we can just let `--quiet` win.